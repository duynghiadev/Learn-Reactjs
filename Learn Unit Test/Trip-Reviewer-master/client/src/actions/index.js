import axios from 'axios';

import {
  GET_TRIPS,
  GET_USERS,
  GET_TRIP_W_REVIEWER,
  CLEAR_TRIP_W_REVIEWER,
  GET_TRIP_W_REVIEWER_ERROR,
  USER_REGISTER,
  USER_LOGIN,
  RESET_PASSWORD,
  USER_AUTH,
  ADD_TRIP,
  UPDATE_TRIP,
  UPDATE_USER,
  DELETE_TRIP,
  GET_TRIP,
  GET_TRIP_ERROR,
  CLEAR_NEW_TRIP,
  CLEAR_TRIP,
  GET_USER_REVIEWS,
} from '../constants/action-names';

export function getTrips(limit = 10, start = 0, order = 'asc', list = []) {
  const request = axios.get(`/api/getManyTrips?limit=${limit}&skip=${start}&order=${order}`);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: GET_TRIPS,
        payload: {
          list: [...list, ...response.data],
          newTripsCount: response.data.length,
        },
      });
    });
}

export function getTripWithReviewer(id) {
  const request = axios.get(`/api/getTrip?id=${id}`);

  return dispatch =>
    request
      .then(({ data: tripData }) => {
        const trip = tripData;

        return axios.get(`/api/getReviewer?id=${trip.ownerId}`).then(({ data: reviewerData }) => {
          const response = {
            trip,
            reviewer: reviewerData,
          };

          dispatch({
            type: GET_TRIP_W_REVIEWER,
            payload: response,
          });
        });
      })
      .catch(error =>
        dispatch({
          type: GET_TRIP_W_REVIEWER_ERROR,
          payload: error,
        })
      );
}

export function getUserReviews(userId) {
  const request = axios.get(`/api/getUserReviews?user=${userId}`);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: GET_USER_REVIEWS,
        payload: response.data,
      });
    });
}

export function updateUser(data) {
  const request = axios.post('/api/userUpdate', data);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: UPDATE_USER,
        payload: response.data,
      });
    });
}

export function getTrip(id) {
  const request = axios.get(`/api/getTrip?id=${id}`);

  return dispatch =>
    request
      .then(response => {
        dispatch({
          type: GET_TRIP,
          payload: response.data,
        });
      })
      .catch(error => {
        dispatch({
          type: GET_TRIP_ERROR,
          payload: error,
        });
      });
}

export function updateTrip(data) {
  const request = axios.post('/api/tripUpdate', data);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: UPDATE_TRIP,
        payload: response.data,
      });
    });
}

export function deleteTrip(id) {
  const request = axios.delete(`/api/tripDelete?id=${id}`);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: DELETE_TRIP,
        payload: response.data,
      });
    });
}

export function clearTrip() {
  return {
    type: CLEAR_TRIP,
    payload: {
      trip: null,
      updatedTrip: false,
      postDeleted: false,
    },
  };
}

export function clearTripWithReviewer() {
  return {
    type: CLEAR_TRIP_W_REVIEWER,
    payload: {
      trip: null,
      reviewer: null,
    },
  };
}

export function loginUser({ email, password }) {
  const request = axios.post('/api/login', { email, password });

  return dispatch =>
    request.then(response => {
      dispatch({
        type: USER_LOGIN,
        payload: response.data,
      });
    });
}

export function resetPassword({ email }) {
  const request = axios.post('/api/resetPassword', { email });

  return dispatch =>
    request.then(response => {
      dispatch({
        type: RESET_PASSWORD,
        payload: response.data,
      });
    });
}

export function auth() {
  const request = axios.get('/api/auth');

  return dispatch =>
    request.then(response => {
      dispatch({
        type: USER_AUTH,
        payload: response.data,
      });
    });
}

export function addTrip(trip) {
  const request = axios.post('/api/trip', trip);

  return dispatch =>
    request.then(response => {
      dispatch({
        type: ADD_TRIP,
        payload: response.data,
      });
    });
}

export function clearNewTrip() {
  return {
    type: CLEAR_NEW_TRIP,
    payload: null,
  };
}

export function getUsers() {
  const request = axios.get('/api/users');

  return dispatch =>
    request.then(response => {
      dispatch({
        type: GET_USERS,
        payload: response.data,
      });
    });
}

export function userRegister(user) {
  const request = axios.post('/api/register', user);

  return dispatch =>
    request.then(({ data }) => {
      dispatch({
        type: USER_REGISTER,
        payload: data,
      });
    });
}
