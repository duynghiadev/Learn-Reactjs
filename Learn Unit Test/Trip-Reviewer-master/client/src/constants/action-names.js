export const GET_TRIPS = 'get-trips';
export const GET_USERS = 'get-users';
export const ADD_TRIP = 'add-trip';
export const UPDATE_TRIP = 'update-trip';
export const UPDATE_USER = 'update-user';
export const DELETE_TRIP = 'delete-trip';
export const GET_TRIP = 'get-trip';
export const GET_TRIP_ERROR = 'get-trip-error';
export const GET_TRIP_W_REVIEWER = 'get-trip-w-reviewer';
export const GET_TRIP_W_REVIEWER_ERROR = 'get-trip-w-reviewer-error';
export const CLEAR_TRIP_W_REVIEWER = 'clear-trip-w-reviewer';
export const CLEAR_NEW_TRIP = 'clear-new-trip';
export const CLEAR_TRIP = 'clear-trip';
export const USER_REGISTER = 'user-register';
export const USER_LOGIN = 'user-login';
export const RESET_PASSWORD = 'reset-password';
export const USER_AUTH = 'user-auth';
export const GET_USER_REVIEWS = 'get-user-reviews';
