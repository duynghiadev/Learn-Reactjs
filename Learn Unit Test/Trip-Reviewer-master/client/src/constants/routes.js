export const HOME = '/';
export const LOGIN = '/login';
export const RESET = '/reset-password';
export const LOGOUT = '/user/logout';
export const REGISTER = '/user/register';
export const USER_PROFILE = '/user';
export const USER_REVIEWS = '/user/user-reviews';
export const EDIT_PROFILE = '/user/edit-profile';
export const EDIT_POST = '/user/edit-post';
export const ADD_TRIP = '/trip/add';
export const TRIPS = '/trips';
