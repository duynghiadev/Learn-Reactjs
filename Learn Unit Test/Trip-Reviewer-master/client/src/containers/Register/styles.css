.register-container {
  text-align: center;
  display: flex;
  flex-direction: row;
  justify-content: space-evenly;
}
.register-container .form-container {
  margin: 0 25px 0 100px;
}
.register-container .user-list-container {
  margin: 0 100px 0 25px;
}
.register-container .form_element {
  margin-top: 20px;
}
.register-container .form-title {
  font-size: 33px;
  font-weight: 300;
  color: #02bcd4;
}
.register-container input {
  font-size: 20px;
  border-radius: 3px;
  border: 1px solid #ebebeb;
  box-sizing: border-box;
  padding: 11px 31px;
  background: #f7f7f7;
  font-weight: 300;
  width: 260px;
}
.register-container button {
  font-size: 20px;
  padding: 10px 52px;
  background: #02bcd4;
  color: #fff;
  font-weight: 300;
  border-radius: 3px;
  border: none;
  margin-top: 20px;
  width: 260px;
  cursor: pointer;
}
.register-container button:hover {
  transform: scale(1.05);
  background: #009eb3;
}
.register-container button:active {
  transform: scale(1.01);
  background: #0495a8;
}
.register-container .error {
  color: #ff5722;
  margin-top: 5px;
}
.register-container .field-error {
  border: 2px solid #ff5722;
  color: #ff5722;
}
.register-container .users-title {
  margin-bottom: 13px;
  font-size: 33px;
  font-weight: 300;
  color: orange;
}
.register-container .existing-users {
  display: flex;
  flex-flow: wrap;
  justify-content: space-evenly;
  margin: 0 auto 50px auto;
}
.register-container .user-container {
  margin: 15px;
}
.register-container .user-avatar {
  width: 150px;
  height: 150px;
}
.register-container .user-names {
  width: 150px;
  margin-top: 10px;
}
.register-container .user-names p {
  color: #404040;
  font-size: 18px;
  font-weight: bold;
  margin: 3px;
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}

@media (max-width: 1049px) {
  .register-container {
    flex-direction: column;
  }
  .register-container .form-container {
    margin: 0;
  }
  .register-container .user-list-container {
    margin: 50px;
  }
}

@media (max-width: 600px) {
  .register-container {
    text-align: center;
  }
  .register-container .user-list-container {
    margin: 10px 25px;
  }
  .register-container .users-title {
    margin: 50px auto 25px auto;
  }
  .register-container .existing-users {
    display: block;
  }
  .register-container .user-container {
    border: 1px solid #ebebeb;
    background-color: #f7f7f7;
    display: flex;
    padding: 15px;
    margin: 5px 0;
  }
  .register-container .user-avatar {
    width: 50px;
    height: 50px;
  }
  .register-container .user-names {
    width: calc(100% - 75px);
    margin: 3px 0 0 25px;
    text-align: left;
  }
  .register-container .user-names p {
    margin: 0;
  }
}
