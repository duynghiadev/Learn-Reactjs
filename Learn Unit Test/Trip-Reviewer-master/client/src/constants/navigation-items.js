import * as routes from './routes';

export const home = 'HOME';
export const userProfile = 'USER_PROFILE';
export const signup = 'SIGNUP';
export const login = 'LOGIN';
export const userReviews = 'USER_REVIEWS';
export const addTrip = 'ADD_TRIP';
export const logout = 'LOGOUT';

const items = {
  [home]: {
    id: 1,
    icon: 'home',
    text: 'Home',
    link: routes.HOME,
    restricted: false,
  },
  [userProfile]: {
    id: 2,
    icon: 'user',
    text: 'Profile',
    link: routes.USER_PROFILE,
    restricted: true,
  },
  [signup]: {
    id: 3,
    icon: 'user-plus',
    text: 'Signup',
    link: routes.REGISTER,
    restricted: false,
    exclude: true,
  },
  [login]: {
    id: 4,
    icon: 'sign-in-alt',
    text: 'Login',
    link: routes.LOGIN,
    restricted: false,
    exclude: true,
  },
  [userReviews]: {
    id: 5,
    icon: 'edit',
    text: 'Edit',
    link: routes.USER_REVIEWS,
    restricted: true,
  },
  [addTrip]: {
    id: 6,
    icon: 'file-alt',
    text: 'Add',
    link: routes.ADD_TRIP,
    restricted: true,
  },
  [logout]: {
    id: 7,
    icon: 'sign-out-alt',
    text: 'Logout',
    link: routes.LOGOUT,
    restricted: true,
  },
};

export default items;
