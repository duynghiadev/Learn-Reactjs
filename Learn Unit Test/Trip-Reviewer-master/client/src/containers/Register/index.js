import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import UserAvatar from '../../components/UserAvatar';
import { getUsers, userRegister } from '../../actions';
import { HOME } from '../../constants/routes';

import './styles.css';

class Register extends PureComponent {
  state = {
    formData: {
      name: '',
      lastname: '',
      email: '',
      password: '',
    },
    hideError: {
      name: false,
      lastname: false,
      email: false,
      password: false,
    },
  };

  componentWillMount() {
    this.props.dispatch(getUsers());
  }

  componentWillReceiveProps(nextProps) {
    const { login } = nextProps.users;

    if (login.isAuth) {
      this.props.history.push(HOME);
    } else {
      this.setState({
        hideError: {
          name: false,
          lastname: false,
          email: false,
          password: false,
        },
      });
    }
  }

  getErrorClass(fieldName) {
    return this.formFieldHasError(fieldName) ? 'field-error' : '';
  }

  handleInput = event => {
    const newFormData = { ...this.state.formData };
    const newHideError = { ...this.state.hideError };
    const { value, name } = event.target;

    newFormData[name] = value;
    newHideError[name] = true;

    this.setState({
      formData: newFormData,
      hideError: newHideError,
    });
  };

  submitForm = event => {
    event.preventDefault();

    this.props.dispatch(userRegister(this.state.formData));
  };

  formFieldHasError(fieldName) {
    const { register } = this.props.users;
    const { hideError } = this.state;
    const errors = register && register.error && register.error.errors;

    return errors && errors[fieldName] && !hideError[fieldName];
  }

  renderError(fieldName) {
    if (this.formFieldHasError(fieldName)) {
      const error = this.props.users.register.error.errors[fieldName];

      return <div className="error">{error.message}</div>;
    }
    return null;
  }

  renderForm() {
    const { name, lastname, email, password } = this.state.formData;

    return (
      <div className="form-container">
        <form onSubmit={this.submitForm}>
          <h2 className="form-title">Sing Up</h2>

          <div className="form_element">
            <input
              type="text"
              className={this.getErrorClass('name')}
              placeholder="Enter name"
              value={name}
              name="name"
              onChange={this.handleInput}
            />
          </div>
          {this.renderError('name')}

          <div className="form_element">
            <input
              type="text"
              className={this.getErrorClass('lastname')}
              placeholder="Enter Lastname"
              value={lastname}
              name="lastname"
              onChange={this.handleInput}
            />
          </div>
          {this.renderError('lastname')}

          <div className="form_element">
            <input
              type="email"
              className={this.getErrorClass('email')}
              placeholder="Enter Email"
              value={email}
              name="email"
              onChange={this.handleInput}
            />
          </div>
          {this.renderError('email')}

          <div className="form_element">
            <input
              type="password"
              className={this.getErrorClass('password')}
              placeholder="Enter Password"
              value={password}
              name="password"
              onChange={this.handleInput}
            />
          </div>
          {this.renderError('password')}

          <button type="submit">Sign Up</button>
        </form>
      </div>
    );
  }

  renderUsers() {
    const { users } = this.props.users;

    if (users && users.length) {
      return (
        <div className="user-list-container">
          <h2 className="users-title">Existing Users:</h2>
          <div className="existing-users limited-width">
            {users.map(user => (
              <div key={user._id} className="user-container">
                <div className="user-avatar">
                  <UserAvatar filename={user.avatar} />
                </div>
                <div className="user-names">
                  <p title={user.name}>{user.name}</p>
                  <p title={user.lastname}>{user.lastname}</p>
                </div>
              </div>
            ))}
          </div>
        </div>
      );
    }

    return null;
  }

  render() {
    return (
      <div className="register-container">
        {this.renderForm()}
        {this.renderUsers()}
      </div>
    );
  }
}

Register.propTypes = {
  users: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
  dispatch: PropTypes.func.isRequired,
};

function mapStateToProps(state) {
  return {
    users: state.users,
  };
}

export default connect(mapStateToProps)(Register);
