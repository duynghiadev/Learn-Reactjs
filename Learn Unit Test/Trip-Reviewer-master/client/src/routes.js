import React from 'react';
import { Switch, Route } from 'react-router-dom';

import NotFound from './components/404';
import Home from './containers/Home';
import User from './components/User';
import EditUserProfile from './containers/EditUserProfile';
import Logout from './components/Logout';
import Login from './containers/Login';
import ResetPass from './containers/ResetPass';
import Register from './containers/Register';
import AddTrip from './containers/AddTrip';
import EditTrip from './containers/EditTrip';
import TripView from './containers/TripView';
import UserReviews from './containers/UserReviews';

import Layout from './hoc/layout';
import Auth from './hoc/auth';

import * as routes from './constants/routes';

const Routes = () => (
  <Layout>
    <Switch>
      <Route path={routes.HOME} exact component={Auth(Home, null)} />
      <Route path={routes.LOGIN} exact component={Auth(Login, false)} />
      <Route path={routes.RESET} exact component={Auth(ResetPass, false)} />
      <Route path={routes.LOGOUT} exact component={Auth(Logout, true)} />
      <Route path={routes.REGISTER} exact component={Auth(Register, false)} />
      <Route path={routes.USER_PROFILE} exact component={Auth(User, true)} />
      <Route path={routes.USER_REVIEWS} exact component={Auth(UserReviews, true)} />
      <Route path={`${routes.EDIT_PROFILE}`} exact component={Auth(EditUserProfile, true)} />
      <Route path={`${routes.EDIT_POST}/:id`} exact component={Auth(EditTrip, true)} />
      <Route path={routes.ADD_TRIP} exact component={Auth(AddTrip, true)} />
      <Route path={`${routes.TRIPS}/:id`} exact component={Auth(TripView, null)} />
      <Route path="*" component={Auth(NotFound, null)} />
    </Switch>
  </Layout>
);

export default Routes;
