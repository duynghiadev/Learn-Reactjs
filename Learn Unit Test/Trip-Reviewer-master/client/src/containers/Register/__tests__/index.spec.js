import React from 'react';
import configureStore from 'redux-mock-store';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';

import Register from '..';
import { userRegister } from '../../../actions';
import { HOME } from '../../../constants/routes';

jest.mock('../../../components/UserAvatar', () => 'UserAvatar');
jest.mock('./../../../actions', () => ({
  getUsers: jest.fn(() => ({
    type: 'getUsers',
  })),
  userRegister: jest.fn(() => ({
    type: 'userRegister',
  })),
}));

const mockStore = configureStore();

const mockComponent = (initialState = {}, props) => {
  const store = mockStore(initialState);

  return <Register {...props} store={store} />;
};

describe('<Register />', () => {
  it('should render component', () => {
    const initialState = {
      users: {},
    };
    const tree = create(mockComponent(initialState)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with users', () => {
    const initialState = {
      users: {
        users: [
          {
            _id: '_id',
            name: 'name',
            lastname: 'lastname',
          },
        ],
      },
    };
    const tree = create(mockComponent(initialState)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render component with errors', () => {
    const initialState = {
      users: {
        register: {
          error: {
            errors: {
              name: { message: 'Test name' },
              lastname: { message: 'Test lastname' },
              email: { message: 'Test email' },
              password: { message: 'Test password' },
            },
          },
        },
      },
    };
    const tree = create(mockComponent(initialState)).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should change state when handleInput is called', () => {
    const initialState = {
      users: {},
    };
    const event = {
      target: {
        value: 'test',
        name: 'email',
      },
    };

    const instance = shallow(mockComponent(initialState)).dive().instance();

    instance.handleInput(event);

    expect(instance.state.formData.email).toEqual('test');
    expect(instance.state.hideError.email).toEqual(true);
  });

  it('should dispatch userRegister when submitForm is called', () => {
    const preventDefault = jest.fn();
    const state = {
      formData: {
        name: 'name',
        lastname: 'lastname',
        email: 'email',
        password: 'password',
      },
    };
    const initialState = {
      users: {},
    };
    const event = { preventDefault };

    const instance = shallow(mockComponent(initialState)).dive().instance();
    instance.setState({ ...state });

    instance.submitForm(event);

    expect(preventDefault).toHaveBeenCalled();
    expect(userRegister).toHaveBeenCalledWith(state.formData);
  });

  it('should redirect when componentWillReceiveProps is called and authenticated', () => {
    const push = jest.fn();
    const props = {
      history: {
        push,
      },
    };
    const nextProps = {
      users: {
        login: {
          isAuth: true,
        },
      },
    };
    const initialState = {
      users: {},
    };

    const instance = shallow(mockComponent(initialState, props)).dive().instance();

    instance.componentWillReceiveProps(nextProps);

    expect(push).toHaveBeenCalledWith(HOME);
  });

  it('should not redirect and should set state when componentWillReceiveProps is called and not authenticated', () => {
    const push = jest.fn();
    const state = {
      hideError: {
        name: true,
        lastname: true,
        email: true,
        password: true,
      },
    };
    const props = {
      history: {
        push,
      },
    };
    const nextProps = {
      users: {
        login: {
          isAuth: false,
        },
      },
    };
    const initialState = {
      users: {},
    };

    const instance = shallow(mockComponent(initialState, props)).dive().instance();
    instance.setState({ ...state });

    instance.componentWillReceiveProps(nextProps);

    expect(push).not.toHaveBeenCalledWith(HOME);
    expect(instance.state.hideError.name).toEqual(false);
    expect(instance.state.hideError.lastname).toEqual(false);
    expect(instance.state.hideError.email).toEqual(false);
    expect(instance.state.hideError.password).toEqual(false);
  });
});
