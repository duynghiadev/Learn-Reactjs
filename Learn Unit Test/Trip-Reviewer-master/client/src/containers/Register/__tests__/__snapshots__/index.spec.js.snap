// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Register /> should render component 1`] = `
<div
  className="register-container"
>
  <div
    className="form-container"
  >
    <form
      onSubmit={[Function]}
    >
      <h2
        className="form-title"
      >
        Sing Up
      </h2>
      <div
        className="form_element"
      >
        <input
          className=""
          name="name"
          onChange={[Function]}
          placeholder="Enter name"
          type="text"
          value=""
        />
      </div>
      <div
        className="form_element"
      >
        <input
          className=""
          name="lastname"
          onChange={[Function]}
          placeholder="Enter Lastname"
          type="text"
          value=""
        />
      </div>
      <div
        className="form_element"
      >
        <input
          className=""
          name="email"
          onChange={[Function]}
          placeholder="Enter Email"
          type="email"
          value=""
        />
      </div>
      <div
        className="form_element"
      >
        <input
          className=""
          name="password"
          onChange={[Function]}
          placeholder="Enter Password"
          type="password"
          value=""
        />
      </div>
      <button
        type="submit"
      >
        Sign Up
      </button>
    </form>
  </div>
</div>
`;

exports[`<Register /> should render component with errors 1`] = `
<div
  className="register-container"
>
  <div
    className="form-container"
  >
    <form
      onSubmit={[Function]}
    >
      <h2
        className="form-title"
      >
        Sing Up
      </h2>
      <div
        className="form_element"
      >
        <input
          className="field-error"
          name="name"
          onChange={[Function]}
          placeholder="Enter name"
          type="text"
          value=""
        />
      </div>
      <div
        className="error"
      >
        Test name
      </div>
      <div
        className="form_element"
      >
        <input
          className="field-error"
          name="lastname"
          onChange={[Function]}
          placeholder="Enter Lastname"
          type="text"
          value=""
        />
      </div>
      <div
        className="error"
      >
        Test lastname
      </div>
      <div
        className="form_element"
      >
        <input
          className="field-error"
          name="email"
          onChange={[Function]}
          placeholder="Enter Email"
          type="email"
          value=""
        />
      </div>
      <div
        className="error"
      >
        Test email
      </div>
      <div
        className="form_element"
      >
        <input
          className="field-error"
          name="password"
          onChange={[Function]}
          placeholder="Enter Password"
          type="password"
          value=""
        />
      </div>
      <div
        className="error"
      >
        Test password
      </div>
      <button
        type="submit"
      >
        Sign Up
      </button>
    </form>
  </div>
</div>
`;

exports[`<Register /> should render component with users 1`] = `
<div
  className="register-container"
>
  <div
    className="form-container"
  >
    <form
      onSubmit={[Function]}
    >
      <h2
        className="form-title"
      >
        Sing Up
      </h2>
      <div
        className="form_element"
      >
        <input
          className=""
          name="name"
          onChange={[Function]}
          placeholder="Enter name"
          type="text"
          value=""
        />
      </div>
      <div
        className="form_element"
      >
        <input
          className=""
          name="lastname"
          onChange={[Function]}
          placeholder="Enter Lastname"
          type="text"
          value=""
        />
      </div>
      <div
        className="form_element"
      >
        <input
          className=""
          name="email"
          onChange={[Function]}
          placeholder="Enter Email"
          type="email"
          value=""
        />
      </div>
      <div
        className="form_element"
      >
        <input
          className=""
          name="password"
          onChange={[Function]}
          placeholder="Enter Password"
          type="password"
          value=""
        />
      </div>
      <button
        type="submit"
      >
        Sign Up
      </button>
    </form>
  </div>
  <div
    className="user-list-container"
  >
    <h2
      className="users-title"
    >
      Existing Users:
    </h2>
    <div
      className="existing-users limited-width"
    >
      <div
        className="user-container"
      >
        <div
          className="user-avatar"
        >
          <UserAvatar
            filename={undefined}
          />
        </div>
        <div
          className="user-names"
        >
          <p
            title="name"
          >
            name
          </p>
          <p
            title="lastname"
          >
            lastname
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
